packages/ember-routing/lib/location/hash_location.js

```diff
@@ -106,9 +104,7 @@ export default EmberObject.extend({
     @param callback {Function}
   */
   onUpdateURL(callback) {
-    let guid = guidFor(this);
-
-    jQuery(window).on(`hashchange.ember-location-${guid}`, () => {
+    this._hashchangeHandler = () => {
```

Comment:
@rwjblue actually, it might be predunt to an an assert that yells if the function is invoked if `this._hashchangeHandler` has already been set, or if its been invoked more then once.